### ISSUE TITLE
ARKit broken after loading new scene.

### ISSUE THREAD

**Issue by sam598** [Unity-Technologies/arfoundation-samples#217]
Steps to reproduce:

1. Open the "SimpleAR" sample scene.
2. Add a button that triggers the code `SceneManager.LoadScene("SimpleAR");`
3. Build and run the project.
4. ARKit will run in the first loaded scene.
5. After reloading the scene ARKit will not start.
6. The button that triggers `ARSession.Reset()` will not work either.

ARKit will not start after loading any other ARFoundation sample scene. There are no errors in Xcode. The app does not crash. The app must be force quit in order to use ARKit again. 

Device: iPad Mini 5
Unity: 2019.1.2f1
OS: iPadOS 13 Beta 3 + 4
ARFoundation: 2.2.0 preview.3
ARKit XR Plugin: 2.2.0 preview.3
ARFoundation Samples: 07/19/2019

**Comment by tdmowrer** [comment]
Sounds like you're loading the same scene twice, which would give you duplicates of the ARSession and the AR managers. These are effectively singletons (though there's nothing preventing multiple instances).

Either unload the AR scene before loading another one, or have a separate scene that just controls the AR lifecycle and load/unload that to turn AR on and off.

**Comment by sam598** [comment]
ARKit will not start after loading any other ARFoundation sample scene.

SimpleAR was used and example, but going from SimpleAR to HumanSegmentationImages and back will have the same result.

This behavior was not present in 2.2.0 preview.1

**Comment by sam598** [comment]
The same behavior happens when loading a scene like SimpleAR, then loading another scene with no ARSession, then loading SimpleAR or any other example scene again.

It's not loading the scene additively so there should only be one ARSession per scene.

**Comment by samuelmorais** [comment]
I am having the same problem. If I reload a scene, switch scene or do any kind of scene changing, the AR Session does not start and the screen is turned black.

**Comment by sam598** [comment]
This issue is still present in:

ARKit XR Plugin 2.2.0 preview.4
AR Foundation 2.2.0 preview.3
ARKit Face Tracking 1.1.0 preview.4

Rolling back to:

ARKit XR Plugin 2.2.0 preview.1
AR Foundation 2.2.0 preview.2
ARKit Face Tracking 1.1.0 preview.1

Works as expected. 

This bug seems like a showstopper for any games with more than one scene. Are there any other suggestions?

**Comment by bambamyi** [comment]
I can also confirm this bug. I have a loader scene that loads the AR scene when the AR scene is done, I unload the scene completely and load it again and the background is black. I am using 2019.2 and 2.2.0 ARFoundation, ARSubsystem, and ARCore. A fix would be much apprecaited.

**Comment by nikolahitek** [comment]
Confirming, I have two AR scenes behavior as explained above. It worked with the previous ARFoundation preview with no problems. 
@tdmowrer Have you guys pinpointed the issue?

**Comment by Rick7221** [comment]
I can confirm this behaviour.

**Comment by sam598** [comment]
Appears to be fixed in AR Foundation 3.0

**Comment by alto971** [comment]
Still have this bug with AR Fundation for Android version !! Even if i go to one NON-AR scene (menu), and then go to a new AR scene ... i've got the black screen bug !!
Have you all resolve this issue with 3.O version of AR Fundation ?

**Comment by MarioUnity** [comment]
@alto971 We have a known issue where switching between AR scenes in 2019.3 causes a black screen, but not going when going from a non-AR scene to an AR Scene. Could you provide the following details so we can investigate?

Device
OS
Unity Version
ARFoundation and AR Kit version

**Comment by alto971** [comment]
@MarioUnity i've been working on another ARFoundation project .. still the same behavior ! 
Menu Scene (non-AR) => AR SCene : OK => return to Menu => another ARScene or the same : KO   We see the camera but the AR exeperience is broken ...seems tracking is off.

**Comment by mdurand42** [comment]
> @MarioUnity i've been working on another ARFoundation project .. still the same behavior !
> Menu Scene (non-AR) => AR SCene : OK => return to Menu: OK => got to another ARScene or the same ARScene: KO We see the camera (no black screen) but the AR experience is broken ...seems tracking is off but no error in console.
> (Unity 2019.4 + ARFoundation and ARCore XR plugin 4.0.8)

@alto971 do you see this issue when switching between scenes in the AR Foundation Samples? We test switching between AR and non-AR scenes all the time and don't see a problem.

-Mike

**Comment by alto971** [comment]
@mdurand42 I'll test with 2 samples and a menu next week, and i'll tell you ! But when you're saying that you're not seing issues .. are you sure it's in the same cases as me ? 
=> Menu Scene (non-AR) => AR SCene : OK => return to Menu: OK => got to another ARScene or the same ARScene: KO We see the camera (no black screen) but the AR experience is broken

**Comment by horatiu665** [comment]
Still a problem. Load Menu -> Load AR scene first time, everything works fine -> Load menu again -> load AR scene second time, the ARSession cannot be .Reset() and no planes show up, however camera still works. 
ARfoundation version 5.0.2 
Unity version 2021.3.8f1 (yeah, I know, but I can't update to unity 2022 cause it's too buggy)
Android - pixel 2 XL
built from macOS M1

**Comment by horatiu665** [comment]
I have tried more and it finally works. I compared it to the setup from the ARFoundation samples project, we had an identical setup but it worked in the samples so it must have been from our side.

I don't know what was the issue really, but might have been something to do with a scene loading plugin we are using (OpenProjects scene loading built-in stuff), or some errors from the ARPointCloudManager being disabled, or something else entirely....!??!! We also use addressables, and the scenes were addressable and placed in the wrong folders (altho they would be loaded correctly) so we fixed that. We also use this Android Resolver thing which has settings that were not shared, so in my case it deleted a lot of android .jar files. I went to Assets > External Dependency Manager > Android Resolver > Settings, disabled Auto-resolution both in editor and build, then Resolved manually, which added the .jar files again, and I'm not sure but that might be what fixed it. We are also using Mapbox which caused its own share of problems.

Anyway, false alarm everyone, carry on. Hope this helps someone.
